File: 0070-climbing-stairs/0070-climbing-stairs.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    let N = n+1
  if (N === 0) return 1;
  if (N === 1) return 1;

  let prev = 0;
  let curr = 1;

  for (let i = 2; i <= N; i++) {
    let next = prev + curr;
    prev = curr;
    curr = next;
  }

  return curr;
};

File: 0070-climbing-stairs/0070-climbing-stairs.cs
public class Solution {
    public int ClimbStairs(int n) {
        int prev = 0;
        int curr = 1;
        for (int i = 2; i <= n+1; i++){
            int next = prev + curr;
            prev = curr;
            curr = next;
        } 
        return curr;
    }
}